src/hooks/useFilterPossiblyMaliciousPositions.ts

```diff
@@ -0,0 +1,58 @@
+import { Interface } from '@ethersproject/abi'
+import { Token } from '@uniswap/sdk-core'
+import ERC20ABI from 'abis/erc20.json'
+import { Erc20Interface } from 'abis/types/Erc20'
+import { useMultipleContractSingleData } from 'lib/hooks/multicall'
+import { useMemo } from 'react'
+import { PositionDetails } from 'types/position'
+import { hasURL } from 'utils/urlChecks'
+
+import { useDefaultActiveTokens } from './Tokens'
+
+const ERC20Interface = new Interface(ERC20ABI) as Erc20Interface
+
+export function useFilterPossiblyMaliciousPositions(positions: PositionDetails[]): PositionDetails[] {
+  const tokenList = useDefaultActiveTokens()
+
+  const nonListPositionTokenAddresses = useMemo(
+    () =>
+      Array.from(
+        new Set(positions.reduce((acc, position) => acc.concat(position.token0, position.token1), [] as string[]))
+      ).filter((address) => !tokenList[address]),
+    [positions, tokenList]
+  )
+
+  const results = useMultipleContractSingleData(nonListPositionTokenAddresses, ERC20Interface, 'symbol')
+  const addressesToSymbol: Record<string, string | undefined> = useMemo(
+    () =>
+      nonListPositionTokenAddresses.reduce(
+        (acc, address, i) => ({
+          ...acc,
+          [address]: results[i].result as string | undefined,
+        }),
+        {}
+      ),
+    [nonListPositionTokenAddresses, results]
+  )
+  return useMemo(
+    () =>
+      positions.filter((position) => {
+        const token0FromList = tokenList[position.token0] as Token | undefined
+        const token1FromList = tokenList[position.token1] as Token | undefined
+        const bothTokensInList = token0FromList && token1FromList
+        if (bothTokensInList) return true
+
+        const token0IsInList = !!token0FromList
+        const token1IsInList = !!token1FromList
+        const atLeastOneTokenIsInList = token0IsInList || token1IsInList
+        const token0HasUrlSymbol = hasURL(token0IsInList ? token0FromList.symbol : addressesToSymbol[position.token0])
+        const token1HasUrlSymbol = hasURL(token1IsInList ? token1FromList.symbol : addressesToSymbol[position.token1])
+        const maxOneUrlTokenSymbol = token0HasUrlSymbol ? !token1HasUrlSymbol : true
+        if (atLeastOneTokenIsInList && maxOneUrlTokenSymbol) return true
```

Comment:
Correct me if I am wrong, but can this entire `useMemo` be simplified to something like this:
```ts
// tokenSymbols contains both symbols from tokenList and RPC call
positions.filter(({token0, token1}) => {
   // both tokens are in the list
   if (tokenSymbols[token0] && tokenSymbols[token1]) { 
      return true
   }
   
   // none of the tokens are in the list
   if (!tokenSymbols[token0] && !tokenSymbols[token1]) {
      return false
   }

  // at this point "atLeastOneTokenIsInList" or both are in the list
  // I know this is a bit code-smell, but still better than ternary. 
  // "maxOne" - sum of two booleans should be 1
  //  "neitherTokenHasAUrlSymbol" - sum of two booleans will be 0
   if (Number(hasURL(tokenSymbols[token0])) + Number(hasURL(tokenSymbols[token1])) <= 1) {
      return true
   }

  // is false the default? when it can happen?
  return false
})
```

Reply:
not quite

if neither token is in the list, then we don't allow any urls.
if one token is in the list, then we allow one url
if both tokens are in the list, then we allow two urls

re: `Number(hasURL(...)) <=1` I personally don't think relying on type coercion (`Number(true)===1`) is more clear than a standard logic language feature.